Pear/Controller/MessagesViewController.swift

```diff
@@ -0,0 +1,191 @@
+        navigationController?.navigationBar.shadowImage = UIImage()
+        navigationController?.navigationBar.titleTextAttributes = [
+            .font: UIFont.getFont(.medium, size: 24)
+        ]
+
+        backButton.setImage(UIImage(named: "backArrow"), for: .normal)
+        backButton.snp.makeConstraints { make in
+            make.size.equalTo(Constants.Onboarding.backButtonSize)
+        }
+        backButton.addTarget(self, action: #selector(backPressed), for: .touchUpInside)
+        navigationItem.leftBarButtonItem = UIBarButtonItem(customView: backButton)
+
+        subtitleLabel.text = "Keep in touch with each other :)"
+        subtitleLabel.textColor = .greenGray
+        subtitleLabel.font = ._12CircularStdBook
+        view.addSubview(subtitleLabel)
+
+        messagesTableView.register(MessagesTableViewCell.self, forCellReuseIdentifier: MessagesTableViewCell.reuseId)
+        messagesTableView.backgroundColor = .backgroundLightGreen
+        messagesTableView.separatorStyle = .none
+        messagesTableView.dataSource = self
+        messagesTableView.delegate = self
+        view.addSubview(messagesTableView)
+
+        getUserMessages()
+        setupConstraints()
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        navigationController?.interactivePopGestureRecognizer?.delegate = self
+    }
+
+    @objc private func backPressed() {
+        navigationController?.popViewController(animated: true)
+    }
+
+    private func setupConstraints() {
+        subtitleLabel.snp.makeConstraints { make in
+            make.centerX.equalToSuperview()
+            make.top.equalTo(view.safeAreaLayoutGuide.snp.top).offset(10)
+        }
+        messagesTableView.snp.makeConstraints { make in
+            make.top.equalTo(subtitleLabel.snp.bottom).offset(20)
+            make.leading.trailing.bottom.equalToSuperview()
+        }
+    }
+
+    private func getUserMessages() {
+        getMessageMatches(netId: user.netID) { matches in
+            matches.forEach { match in
+                guard let pairNetId = match.pair  else {
+                    print("Unable to get the pair's netid from the match.")
+                    return
+                }
+                self.getMessageUserData(pairNetId: pairNetId) { pair in
+                    let messageUser = MessageUser(netID: pair.netID, firstName: pair.firstName, lastName: pair.lastName, status: match.status, meetingTime: match.meetingTime, profilePictureURL: pair.profilePictureURL ?? "")
+                    self.messageUsers.append(messageUser)
+                    self.reloadMessagesTableView()
+                }
+            }
+        }
+    }
+
+    private func reloadMessagesTableView() {
+        self.timer?.invalidate()
+        self.timer = Timer.scheduledTimer(timeInterval: 0.1, target: self, selector: #selector(reloadTableViewWithTimer), userInfo: nil, repeats: false)
```

Comment:
Long line

Reply:
delete self

Reply:
Instead of an objective c selector, you could use the similar function that takes a `block`